dt-backend/src/model/user/user.service.ts

```diff
@@ -2,13 +2,14 @@ import {Injectable} from '@nestjs/common';
 import { prisma } from '../../main';
 import { User } from '../types/user.type';
 import { Prisma } from '@prisma/client'
+import {UserUpdateDto} from "./dto/user-update.dto";
 
 @Injectable()
 export class UserService {
 
-  async User(userWhereUniqueInput: Prisma.UserWhereUniqueInput): Promise<User> {
+  async User(id: Prisma.UserWhereUniqueInput): Promise<User> {
     return prisma.user.findUnique({
-      where: userWhereUniqueInput,
+      where: id,
```

Comment:
If it's just the `id`, then why not pass `id: number` as a parameter and create the object here: `where: { id }`. imo that would make the code more beautiful, which is also part of my suggestions from last review.

Reply:
so where i think i'm struggling is with the prisma documentation crosses over with this. So `prisma.user.findUnique` is one thing.... but instead of returning that can i return my own type??

Reply:
Yup, you totally can! If you wanna modify the object before returning it, you'd first save the prisma return value, then create your own objet using the prisma data! Beware, you have to use an await infront of the prisma in order to work with the data! Otherwise you'll just get the promise!
Simple Code Example:
```ts
let data = await prisma.user.findUnique({ where: id});
return { username: data.username };
```

The other option, if I understood correctly is to use a custom type, but with the __same__ attributes as the Prisma type! In that case you can simply explicitly redefine the return value of the method!
```ts
async getUser(id: number): Promise<CustomPrismaUser> () {
  return prisma.user.findUnique({ where: { id } });
}
```

Reply:
i was missing some syntax. that's all it was lol